options/options_helper.cc

```diff
@@ -255,6 +263,77 @@ bool ParseVectorCompressionType(
   return true;
 }
 
+// This is to handle backward compatibility, where compaction_options_fifo
```

Comment:
why there's backward compatibility issue, since compaction_options_fifo was not dynamically settable? for options file?

Reply:
This code path is also being used for serializing and deserializing options files. So, I believe, until recently I believe when compaction_options_fifo got serialized to persist to disk, compaction_options_fifo.max_table_files_size's value got written out as "compaction_options_fifo": "<max_table_files_size-value>". 

For example, check this: 
https://github.com/facebook/rocksdb/blob/0552029b5c710c332368ee05c112847f7d47c6b7/options/options_test.cc#L83
and:
https://github.com/facebook/rocksdb/blob/0552029b5c710c332368ee05c112847f7d47c6b7/options/options_test.cc#L180-L181

So, to be backward compatible, and support reading old options files during upgrades, we need to support it. I guess we don't really need to support it when people are setting these options dynamically though.